src/net.c

```diff
@@ -1863,6 +1863,8 @@ handle_request(coap_context_t *context, coap_session_t *session, coap_pdu_t *pdu
       coap_delete_string(uri_path);
       return;
     }
+    else if (response)
```

Comment:
This is one of the cases where I'd like you to put the entire else-branch in between parentheses for clarity:

```c
} else {
  if (respose) ...
}
```

Ideally, the body of the following if-statement also had parentheses.